fis/core/ingest.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functionality relating to S3 upload metadata processing"""
+
+import json
+
+from ghga_event_schemas.pydantic_ import FileUploadValidationSuccess
+from ghga_service_commons.utils.crypt import decrypt
+from nacl.exceptions import CryptoError
+from pydantic import ValidationError
+
+from fis.config import Config
+from fis.core import models
+from fis.ports.inbound.ingest import UploadMetadataProcessorPort
+
+
+class UploadMetadataProcessor(UploadMetadataProcessorPort):
+    """Handler for S3 upload metadata processing"""
+
+    def __init__(self, *, config: Config):
```

Comment:
Since this is a service just used for MVP, this is fine. However, usually the core should not depend on outside Config but config classes can be defined right next to the class consuming it. The main config is then composing all the configs from the core, providers, and protocols.

Reply:
Moved ServiceConfig into ingest, didn't get the intention at first.